### ISSUE TITLE
ERR_INVALID_ARG_TYPE still present after the fix

### ISSUE THREAD

**Issue by Jacek-fstack** [cypress-io/cypress#16223]
https://github.com/cypress-io/cypress/issues/15901
```
Fixed a regression in 7.0.0 that caused the Test Runner to crash with an ERR_INVALID_ARG_TYPE type error. We now correctly detect a utf8 request body with multi-byte Unicode characters. Fixes #15901.
```
It's still present in my tests. I'm using cypress-file-upload plugin and it happens when I want to upload an image file. Crashes the whole run.

```
TypeError [ERR_INVALID_ARG_TYPE] [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer or Uint8Array. Received type number (45)
```

Version 7.2.0

**Comment by iozz** [comment]
Same for me, the runner is still crashing on our test runs.

**Comment by EliasJorgensen** [comment]
This is also still happening for us, when intercepting an image upload. The following code has been working fine on Cypress v6:
```javascript
    it('Can edit resource image', () => {
        cy.intercept({
            method: 'POST',
            url: '**/img'
        }).as('imageUpload')

        cy.get('#upload-input')
            .attachFile('hackerman.jpg')

        cy.get('button[type="submit"]').click()

        cy.wait('@imageUpload').then((interception) => {
            assert.equal(interception.response.statusCode, 200)
        })

        cy.contains('The resource was updated')
    })
```

**Comment by jennifer-shehane** [comment]
We did fix the issue that was reproducible at the time of release, but there are still cases of this failing that we could not address because no one provided a way to reproduce them before the next release. 

We are working on this one now: https://github.com/cypress-io/cypress/issues/15898 where there was a reproducible example provided today in this repo: https://github.com/verheyenkoen/cypress-upload-intercept-problem This may be a duplicate of the issue you're experiencing but **we can't know without a reproducible example**.

**Please provide a fully reproducible example that we can run and see Cypress crashing**.

**Comment by todd-m-kemp** [comment]
I'm still experiencing this problem as well and can provide a fully reproducible example.

*Steps to Reproduce:*

(1)  Run the following test code; the test will pass but continue with step (2) to experience the crash.

```js
context('Cypress Issue', function () {

  beforeEach(function () {

    // These are not real credentials.
    // Please sign up (for free) at www.vidyard.com, verify your email address, then update these accordingly.

    const email = 'test@fake.com'
    const password = 'A123456b!'

    cy.visit('https://secure.vidyard.com/user/sign_in')
    cy.get('#username').type(email)
    cy.get('#password').type(password)
    cy.get('#sign-in').click()
  })

  it('Cypress Crash', function () {
    cy.intercept('PUT', '/uploads/videos/**').as('uploadDuration');
  });
});
```

(2) In the Test Runner, click `New` -> `Upload` and select a video file to upload.

At this point, the Cypress process will crash, producing the following output:

```
TypeError [ERR_INVALID_ARG_TYPE] [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer or Uint8Array. Received type number (0)
    at write_ (_http_outgoing.js:696:11)
    at ClientRequest.write (_http_outgoing.js:661:15)
    at Request.write (/Users/tkemp/Library/Caches/Cypress/7.2.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/@cypress/request/request.js:1496:27)
    at /Users/tkemp/Library/Caches/Cypress/7.2.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/@cypress/request/request.js:546:20
    at Array.forEach (<anonymous>:null:null)
    at end (/Users/tkemp/Library/Caches/Cypress/7.2.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/@cypress/request/request.js:545:23)
    at Immediate._onImmediate (/Users/tkemp/Library/Caches/Cypress/7.2.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/@cypress/request/request.js:578:7)
    at processImmediate (internal/timers.js:461:21)
 {
  code: 'ERR_INVALID_ARG_TYPE'
}
TypeError [ERR_INVALID_ARG_TYPE] [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer or Uint8Array. Received type number (0)
    at write_ (_http_outgoing.js:696:11)
    at ClientRequest.write (_http_outgoing.js:661:15)
    at Request.write (/Users/tkemp/Library/Caches/Cypress/7.2.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/@cypress/request/request.js:1496:27)
    at /Users/tkemp/Library/Caches/Cypress/7.2.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/@cypress/request/request.js:546:20
    at Array.forEach (<anonymous>:null:null)
    at end (/Users/tkemp/Library/Caches/Cypress/7.2.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/@cypress/request/request.js:545:23)
    at Immediate._onImmediate (/Users/tkemp/Library/Caches/Cypress/7.2.0/Cypress.app/Contents/Resources/app/packages/server/node_modules/@cypress/request/request.js:578:7)
    at processImmediate (internal/timers.js:461:21)
```

**Comment by jennifer-shehane** [comment]
@todd-m-kemp I can recreate the issue from your example given. I'm not sure if these are related or not until we investigate, but will send this example to the developer investigating.

**Comment by lugosium** [comment]
same for us

**Comment by jennifer-shehane** [comment]
If you would like to download the pre-release version of Cypress to get this fix. Follow the [instructions here](https://github.com/cypress-io/cypress/commit/5c453115334e290af66b3594a19e4e455cfa7596#commitcomment-50340512)

**Comment by lugosium** [comment]
> If you would like to download the pre-release version of Cypress to get this fix. Follow the [instructions here](https://github.com/cypress-io/cypress/commit/5c453115334e290af66b3594a19e4e455cfa7596#commitcomment-50340512)

Working perfectly 💯, thank you @jennifer-shehane
